full/src/main/java/apoc/dv/JDBCResource.java

```diff
@@ -0,0 +1,94 @@
+package apoc.dv;
+
+import org.apache.commons.lang3.StringUtils;
+import org.neo4j.internal.helpers.collection.Pair;
+
+import java.util.List;
+import java.util.Map;
+import java.util.Objects;
+import java.util.regex.MatchResult;
+import java.util.stream.Collectors;
+import java.util.stream.LongStream;
+import java.util.stream.Stream;
+
+public class JDBCResource extends VirtualizedResource {
+
+    private final String queryParsed;
+
+    public JDBCResource(String name, Map<String, Object> config) {
+        super(Objects.requireNonNull(name, "Field `name` should be defined"),
```

Comment:
is there a reason we did not enforce these in the `CSVResource`?

Reply:
never mind, the checks are actually performed in the parent class and should be removed here